flat_mutation_reader.cc

```diff
@@ -971,6 +971,105 @@ make_flat_mutation_reader_from_fragments(schema_ptr schema, reader_permit permit
     return make_flat_mutation_reader_from_fragments(std::move(schema), std::move(permit), std::move(filtered), pr);
 }
 
+flat_mutation_reader
+make_slicing_filtering_reader(flat_mutation_reader rd, const dht::partition_range& pr, const query::partition_slice& slice) {
+    class reader : public flat_mutation_reader::impl {
+        flat_mutation_reader _rd;
+        const dht::partition_range* _pr;
+        const query::partition_slice* _slice;
+        dht::ring_position_comparator _cmp;
+        std::optional<clustering_ranges_walker> _ranges_walker;
+        std::optional<range_tombstone_splitter> _splitter;
+
+    public:
+        reader(flat_mutation_reader rd, const dht::partition_range& pr, const query::partition_slice& slice)
+            : flat_mutation_reader::impl(rd.schema(), rd.permit())
+            , _rd(std::move(rd))
+            , _pr(&pr)
+            , _slice(&slice)
+            , _cmp(*_schema) {
+        }
+
+        virtual future<> fill_buffer(db::timeout_clock::time_point timeout) override {
+            const auto consume_fn = [this] (mutation_fragment mf) {
+                push_mutation_fragment(std::move(mf));
+            };
```

Comment:
This wrapper adds nothing, AFAICT.

Reply:
I wrote the wrapper because this appears below and looks awkward:
```
_splitter->flush(mf.position(), std::bind(&reader::push_mutation_fragment, this)...
```